pkg/state/state.go

```diff
@@ -1161,9 +1163,9 @@ func (s *stateManager) AddBlock(block []byte) (*proto.Block, error) {
 	return rs, nil
 }
 
-func (s *stateManager) AddDeserializedBlock(block *proto.Block) (*proto.Block, error) {
+func (s *stateManager) AddDeserializedBlock(block *proto.Block, snapshot *proto.BlockSnapshot) (*proto.Block, error) {
 	s.newBlocks.setNew([]*proto.Block{block})
-	rs, err := s.addBlocks()
+	rs, err := s.addBlocks([]*proto.BlockSnapshot{snapshot})
```

Comment:
maybe we should set snapshots just like s.newBlocks.setNew()? Like add snapshots to stateManager struct
`newSnapshots *newSnapshots`
```
type newSnapshots struct {
	curPos         int
	blockSnapshots []*proto.BlockSnapshot
}

func (n *newSnapshots) setNew(blockSnapshots []*proto.BlockSnapshot) {
	n.reset()
	n.blockSnapshots = blockSnapshots
}

func (n *newSnapshots) next() bool {
	n.curPos++
	return n.curPos <= len(n.blockSnapshots)
}

func (n *newSnapshots) current() (*proto.BlockSnapshot, error) {
	if n.curPos > len(n.blockSnapshots) || n.curPos < 1 {
		return nil, errors.New("bad current position")
	}
	blockSnapshot := n.blockSnapshots[n.curPos-1]
	return blockSnapshot, nil
}

func (n *newSnapshots) reset() {
	n.blockSnapshots = nil
	n.curPos = 0
}
```